Adding AI heartbeat check at 200 msec since last heard

patnolan33 opened the pull request.

patnolan33 committed:
Adding AI heartbeat check at 200 msec since last heard

patnolan33 committed:
Adding check to make sure AI messaging has started prior to aborting for lack of heartbeat

kkroeger007 commented:
This is not necessarily the best technique. The solution is relying on the fact that there are other messages that are getting to the autopilot rather than handling the specific logic associated with the AI message. This logic breaks if the fundamental problem is the physical communication link with the aircraft. Additionally, what happens when there is a major delay and a test has restarted. The logic needs to be very careful here and i would recommend also looking at how the normal failsafes are handled. Would like to see an iteration of this and glad to talk it over.

patnolan33 committed:
Moving AI failsafe to be more in line with other failsafes

patnolan33 commented:
Moved AI heartbeat to failsafe actions. Added a check in the mode change logic to reset the AI timestamp to 0 if changing away from AIDF mode. See commit: ba61587a670ea34dc512a7a177d105e7fdc06f97
